test/TEST-24-UNIT-TESTS/test.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+# -*- mode: shell-script; indent-tabs-mode: nil; sh-basic-offset: 4; -*-
+# ex: ts=8 sw=4 sts=4 et filetype=sh
+set -e
+TEST_DESCRIPTION="Run unit tests under containers"
+RUN_IN_UNPRIVILEGED_CONTAINER=yes
+
+. $TEST_BASE_DIR/test-functions
+
+check_result_nspawn() {
+    local _ret=1
+    [[ -e $TESTDIR/$1/testok ]] && _ret=0
+    if [[ -s $TESTDIR/$1/failed ]]; then
+        _ret=$(($_ret+1))
+        echo "=== Failed test log ==="
+        cat $TESTDIR/$1/failed
+    else
+        if [[ -s $TESTDIR/$1/skipped ]]; then
+            echo "=== Skipped test log =="
+            cat $TESTDIR/$1/skipped
+        fi
+        if [[ -s $TESTDIR/$1/testok ]]; then
+            echo "=== Passed tests ==="
+            cat $TESTDIR/$1/testok
+        fi
+    fi
+    [[ -n "$TIMED_OUT" ]] && _ret=$(($_ret+1))
+    return $_ret
+}
+
+test_setup() {
+    if type -P meson && [[ "$(meson configure $BUILD_DIR | grep install-tests | awk '{ print $2 }')" != "true" ]]; then
+        dfatal "Needs to be built with -Dinstall-tests=true"
```

Comment:
I'm not sure the test should fail in this case. Maybe it'd be better to just skip it.

Reply:
Hmm, are there any easy way to skip tests?

Reply:
I think `dwarn ...; exit 0` instead `dfatal ...; exit 1` should be enough to skip the test.

Reply:
I considered that, but it seems not to work with `make setup run`.
I'd like to leave this to another PR...

Reply:
Sure. I have no problem with that.